src/Core/TileMesh.js

```diff
@@ -76,11 +76,11 @@ class TileMesh extends THREE.Mesh {
                 throw new Error('unsupported projection wms for this viewer');
             }
         } else if (source.isTMSSource) {
-            // Special globe case: use the P(seudo)M(ercator) coordinates
+            // Special globe case
             if (CRS.is4326(this.extent.crs) &&
                     (source.extent.crs == 'EPSG:3857' || CRS.is4326(source.extent.crs))) {
-                OGCWebServiceHelper.computeTileMatrixSetCoordinates(this, 'PM');
-                return this.wmtsCoords.PM;
+                OGCWebServiceHelper.computeTileMatrixSetCoordinates(this, source.tileMatrixSet);
```

Comment:
if source is TMS, it must have `tileMatrixSet`, it should not have any difference with `wmtsSource`.
we should have
```js
else if (source.isTMSSource || source.isWTMSSource) {
  OGCWebServiceHelper.computeTileMatrixSetCoordinates(this, source.tileMatrixSet);
  return this.wmtsCoords[source.tileMatrixSet];
}
```

```

Reply:
This is not possible, as this breaks planar cases.

Reply:
`computeTileMatrixSetCoordinates` should return `computeTMSCoordinates`  when `tilematrixset` isn't `PM` or `WGS84`

Reply:
I tried it all, I unfortunately can't do any better :/ There's too much different cases to be supported.

Reply:
You could try a refactoring of `getCoordsForSource`.
It should be replaced by `getExtent(crs)`. The `crs` parameter will be a tiled crs or not.
`getExtent(crs)` could be cached the computed extent in `Set` object.